MetaMask/metamask-extension
# fix: prevent cancel and speedup transactions to call accept approval

## Explanation



Please see the bug ticket here: https://github.com/MetaMask/metamask-extension/issues/18809 
Cancel and Speedup transactions are using the same private method to approve transaction (`approveTransaction`). However, we don't need to accept approvals for cancel and speedup actions. 

## Manual Testing Steps



1. log in to the extension
2. click send button
3. enter an address
4. enter an amount
5. click next
6. confirm the transaction
7. speed up/ cancel the transaction
8. See no errors are thrown in background

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
